CMakeLists.txt

```diff
@@ -128,3 +131,49 @@ endif()
 if(TOKENIZERS_BUILD_TOOLS)
   add_subdirectory(examples/tokenize_tool)
 endif()
+
+# Installation rules
+include(GNUInstallDirs)
+
+# Install the library and its dependencies
+install(
+  TARGETS tokenizers re2 sentencepiece-static
+  EXPORT tokenizers-targets
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+
+# Install header files
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  FILES_MATCHING PATTERN "*.h"
+)
+
+install(
+  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/third-party/json/single_include/
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```

Comment:
Won't this single include also be found in ${CMAKE_CURRENT_SOURCE_DIR}/include/? Then you can update the pattern to also accept `*.hpp`?